google/cloud/ndb/model.py

```diff
@@ -2494,6 +2494,11 @@ def _from_base_type(self, value):
             indicate that the value didn't need to be unwrapped and
             decompressed.
         """
+        # First, check for legacy compressed LocalStructuredProperty values.
+        # See https://github.com/googleapis/python-ndb/issues/359
+        if self._compressed and isinstance(value, ds_entity_module.Entity):
+            return
+
         if self._compressed and not isinstance(value, _CompressedValue):
```

Comment:
I know this isn't part of the PR, but it's drawn my eye to this code, and I have a couple of questions.